### ISSUE TITLE
Wont work with managed identity

### ISSUE THREAD

**Issue by DennisBergemann** [azureautomation/azure-automation-ise-addon#168]
Hi,

iam using for my runbooks a system managed identity. (RunAsAccounts are wrong way)
It seems to be that this way wont work with this cool add-on.

as an example this code can't be used then:
`function Get-GraphAPIAccessToken {
    $resource= "?resource=https://graph.microsoft.com/"
    $url = $env:IDENTITY_ENDPOINT + $resource
    $Headers = New-Object "System.Collections.Generic.Dictionary[[String],[String]]"
    $Headers.Add("X-IDENTITY-HEADER", $env:IDENTITY_HEADER)
    $Headers.Add("Metadata", "True")
    $accessToken = Invoke-RestMethod -Uri $url -Method 'GET' -Headers $Headers
    return $accessToken.access_token
}`

is it a chance to use managed identity with your add-on or see you any way to implement it?

Best Regards
Dennis Bergemann

**Comment by kellywahl** [comment]
Yes I too would really love this feature as it looks like Microsoft are phasing out "Run as" for Runbooks
